### ISSUE TITLE
UnboundLocalError: local variable 'toks' referenced before assignment

### ISSUE THREAD

**Issue by superbobry** [brentp/bwa-meth#6]
Hello, the `master` version seems to be broken after 1eaa967de500f12b617847490f248c439f561093. 

Can you please take a look at [`bwameth.py:291`](https://github.com/brentp/bwa-meth/blob/master/bwameth.py#L291)?

**Comment by mcieslik-mctp** [comment]
I think I am still running into this problem (I updated both bwa-meth and toolshed to the latest github versions).

For some FASTQ pairs the process hangs without an error after some time. I noticed that it typically occurs after a new BAM temp file (i.e 0001) is created.

I have not been able to identify the what the problem with the FASTQ files is (although commas are present in quality scores of files that succeed and not in files that fail. The error is fully reproducible.

```
for aln in handle_reads(pair_list, set_as_failed):
    out.write(str(aln) + '\n')
    <<seems to hang here>>>
```
